gtsam/geometry/Cal3Bundler.cpp

```diff
@@ -99,20 +99,26 @@ Point2 Cal3Bundler::calibrate(const Point2& pi, OptionalJacobian<2, 3> Dcal,
   double x = (pi.x() - u0_) / fx_, y = (pi.y() - v0_) / fx_;
   const Point2 invKPi(x, y);
 
-  // initialize by ignoring the distortion at all, might be problematic for
-  // pixels around boundary
-  Point2 pn(x, y);
+  Point2 pn;
+  double px = pi.x(), py = pi.y();
 
   // iterate until the uncalibrate is close to the actual pixel coordinate
   const int maxIterations = 10;
-  int iteration;
-  for (iteration = 0; iteration < maxIterations; ++iteration) {
-    if (distance2(uncalibrate(pn), pi) <= tol_) break;
-    const double px = pn.x(), py = pn.y(), xx = px * px, yy = py * py;
-    const double rr = xx + yy;
-    const double g = (1 + k1_ * rr + k2_ * rr * rr);
+  int iteration = 0;
+  do {
+    // initialize pn with distortion included
+    double rr = (px * px) + (py * py);
+    double g = (1 + k1_ * rr + k2_ * rr * rr);
```

Comment:
can we retain the `const`?

Reply:
`g` gets updated every iteration so it is not semantically `const`.

Reply:
That's BS ;-)
Make const rr and g const, they are declared inside scope where they are const.

Reply:
Is that how it works? I didn't know `const` was scope-dependent (and haven't seen an example of it anywhere).